lib/connectors/memory.js

```diff
@@ -830,7 +901,9 @@ Memory.prototype.replaceOrCreate = function(model, data, options, callback) {
     // Calling _createSync to update the collection in a sync way and
     // to guarantee to create it in the same turn of even loop
     return self._createSync(model, data, function(err, id) {
-      if (err) return process.nextTick(function() { callback(err); });
+      if (err) return process.nextTick(function() {
```

Comment:
`process.nextTick(callback, err)` while you are at changing this (why is the change needed in the first place?)